matlab/src/cpp/arrow/matlab/array/proxy/chunked_array.cc

```diff
@@ -0,0 +1,187 @@
+        context.outputs[0] = length_mda;
+    }
+
+    void ChunkedArray::getNumChunks(libmexclass::proxy::method::Context& context) {
+        namespace mda = ::matlab::data;
+        mda::ArrayFactory factory;
+        auto length_mda = factory.createScalar(chunked_array->num_chunks());
+        context.outputs[0] = length_mda;
+    }
+
+    void ChunkedArray::getChunk(libmexclass::proxy::method::Context& context) {
+        namespace mda = ::matlab::data;
+        mda::ArrayFactory factory;
+        
+        mda::StructArray args = context.inputs[0];
+        const mda::TypedArray<int32_t> index_mda = args[0]["Index"];
+        const auto matlab_index = int32_t(index_mda[0]);
+        
+        // Note: MATLAB uses 1-based indexing, so subtract 1.
+        // arrow::Schema::field does not do any bounds checking.
+        const int32_t index = matlab_index - 1;
+        const auto num_chunks = chunked_array->num_chunks();
+        
+        if (num_chunks == 0) {
+            context.error = makeEmptyChunkedArrayError();
+            return;
+        }
+        
+        if (matlab_index < 1 || matlab_index > num_chunks) {
+            context.error = makeInvalidNumericIndexError(matlab_index, num_chunks);
+            return;
+        }
+
+        const auto array = chunked_array->chunk(index);
+        MATLAB_ASSIGN_OR_ERROR_WITH_CONTEXT(auto array_proxy,
+                                            arrow::matlab::array::proxy::wrap(array),
+                                            context,
+                                            error::UNKNOWN_PROXY_FOR_ARRAY_TYPE);
+        
+        
+        const auto array_proxy_id = libmexclass::proxy::ProxyManager::manageProxy(array_proxy);
+        const auto type_id = static_cast<int64_t>(array->type_id());
+
+        mda::StructArray output = factory.createStructArray({1, 1}, {"ProxyID", "TypeID"});
+        output[0]["ProxyID"] = factory.createScalar(array_proxy_id);
+        output[0]["TypeID"] = factory.createScalar(type_id);
+        context.outputs[0] = output;
+    }
+
+
+    void ChunkedArray::getType(libmexclass::proxy::method::Context& context) {
+        namespace mda = ::matlab::data;
+
+        mda::ArrayFactory factory;
+
+        MATLAB_ASSIGN_OR_ERROR_WITH_CONTEXT(auto type_proxy,
+                                            type::proxy::wrap(chunked_array->type()),
+                                            context,
+                                            error::ARRAY_FAILED_TO_CREATE_TYPE_PROXY);
+
+
+        const auto proxy_id = libmexclass::proxy::ProxyManager::manageProxy(type_proxy);
+        const auto type_id = static_cast<int32_t>(type_proxy->unwrap()->id());
+
+        mda::StructArray output = factory.createStructArray({1, 1}, {"ProxyID", "TypeID"});
+        output[0]["ProxyID"] = factory.createScalar(proxy_id);
+        output[0]["TypeID"] = factory.createScalar(type_id);
+        context.outputs[0] = output;
```

Comment:
`Array::type()` uses 2 row scalars instead of a struct array: https://github.com/apache/arrow/blob/a43206846a02028fc2ac5091c388aa65b07799c1/matlab/src/cpp/arrow/matlab/array/proxy/array.cc#L93-L94

Which is preferred?

Reply:
I think the `struct` array is preferred. With the two output parameters, you have to know if output 1 is the proxy id or the type id. I can see someone mixing up the order easily. Returning a `struct` array eliminates this risk. 

I'll go ahead and make that change to `Array`.